### PR TITLE
Inline purchase flow

### PR DESCRIPTION
**Pull request by alex-deaconu**
## Description
Inline purchase flow

Move plans and purchase modals inline
Create routes to replace the modal navigation

Update factory methods

[stage-2]

## Motivation and Context
Updated structure for the Billing Friction epic

## How Has This Been Tested?
Tested changes locally.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
No

### CODE REVIEW

**Comment by alex-deaconu** [comment]
This will be the base PR for the inline purchase flow.

@ezequielc please review. Thanks!

File: web/scripts/purchase/controllers/ctr-purchase.js [hunk 1]
```diff
@@ -133,22 +131,22 @@ angular.module('risevision.apps.purchase')
...
 
-              $modalInstance.close('success');
+              $state.go('apps.home');
             }
           });
         }
       };
 
       $scope.dismiss = function () {
-        $modalInstance.dismiss('cancel');
```
Comment (ezequielc) [inline web/scripts/purchase/controllers/ctr-purchase.js:151]: @alex-deaconu Do we still call `dismiss()` or could this be removed? Thanks!

**Review by ezequielc — APPROVED** [review]
@alex-deaconu Code looks good. I will test the functionality properly when reviewing the next PRs. Thanks!

File: web/scripts/purchase/controllers/ctr-purchase.js (hunk 1 above)
Comment (alex-deaconu) [reply web/scripts/purchase/controllers/ctr-purchase.js:151]: @ezequielc I haven't got to this part of the flow yet, but this close/dismiss functionality is definitely on my least to clean up. Thanks!

Comment (ezequielc) [reply web/scripts/purchase/controllers/ctr-purchase.js:151]: @alex-deaconu Sounds good, just wanted to make sure it was covered. 👍 Thanks!

**Comment by alex-deaconu** [comment]
@ezequielc I had also introduced some issues with Storage module initialization, as per failing e2e tests. They should now be fixed 🤞 . Thanks.

**Comment by alex-deaconu** [comment]
@ezequielc as discussed, the last commit handles the case where the User switches companies during the purchase flow. Thanks.

**Pull request merged**